### PR TITLE
Music box demo

### PR DESCRIPTION
**Pull request by jcelerier**
Sounds should be ok - please test. I also added a reverb to make it less bland.

### CODE REVIEW

File: examples/kuesa/music-box/qml/MainScene.qml [hunk 1]
```diff
@@ -0,0 +1,331 @@
+import Qt3D.Core 2.10
```
Comment (lemirep) [inline examples/kuesa/music-box/qml/MainScene.qml]: Missing copyright header

File: examples/kuesa/music-box/qml/main.qml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import QtQuick 2.11
```
Comment (lemirep) [inline examples/kuesa/music-box/qml/main.qml]: Missing copyright header

File: examples/kuesa/music-box/qml/MainScene.qml [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+import Qt3D.Render 2.11
```
Comment (lemirep) [inline examples/kuesa/music-box/qml/MainScene.qml]: Might make sense to use include 2.12 everywhere as we assume Qt 5.12 for Kuesa ?

File: examples/kuesa/music-box/music-box.pro [hunk 4]
```diff
@@ -0,0 +1,44 @@
+TEMPLATE = app
```
Comment (lemirep) [inline examples/kuesa/music-box/music-box.pro]: Missing copyright header

File: examples/kuesa/music-box/main.cpp [hunk 5]
```diff
@@ -0,0 +1,47 @@
+#include <QGuiApplication>
```
Comment (lemirep) [inline examples/kuesa/music-box/main.cpp]: Missing copyright header

**Review by lemirep — COMMENTED** [review]
When running the demo, is it normal that some of the metal stuff that go against the cylinder with spike disappear?

**Comment by jcelerier** [comment]
@mpersano do you know where to look for the metal prongs disappearing ?

**Comment by lemirep** [comment]
I'm still seeing prongs disappearing, is this supposed to be fixed ?

**Comment by jcelerier** [comment]
not yet, sorry

**Comment by jcelerier** [comment]
hmm, with qt3d at commit fed848f7dcff99cf5adb3b7b45190826b3dcf898 the sprongs don't disappear anymore (maybe it was the problem that @jjcasmar fixed with animation interpolation?), however I have some weird artefacts which appeared and follow the motion of the music box handle : 

hmm

**Comment by jcelerier** [comment]
and I don't have the mechanical arm or floor anymore...

**Comment by lemirep** [comment]
Rebased against latest dev, looks good here, could you give it another try please?

**Comment by jcelerier** [comment]
on latest qt3d / kuesa it seems to work fine  - anything else you see in the code ? @zakalawe is the sound ok ?

**Comment by zakalawe** [comment]
@jcelerier  I'm out of the office this week, will try to test this evening but might forget. Otherwise nag me next week.

**Comment by zakalawe** [comment]
I built the branch last night but can't run due to missing assets (QRB) at runtime - this is after doing a git-lfs update so I think the files are present. Is there some install / copying / runtime step needed?

**Comment by jcelerier** [comment]
@zakalawe can you test again on macOS ? I couldn't find a way to reproduce your QRB issue but I fixed another issue when building with clang.

**Comment by jcelerier** [comment]
moved to gerrit

**Pull request closed without merging**